src/plugins/expressions/public/functions/combine_filters.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Filter } from '../../common/expression_types';
+import { ExpressionFunction } from '../../common/types';
+
+interface Arguments {
+  filter: string;
+}
+
+type Output = Filter;
+
+export function combineFilters(): ExpressionFunction<'combineFilters', null, Arguments, Output> {
+  return {
+    name: 'combineFilters',
+    type: 'filter',
+    help: 'takes two filters and merges them',
+    args: {
+      filter: {
+        types: ['filter'],
```

Comment:
The key here is that we still specify `types: ['filter']` (not `string`).

The `filter` type in the types registry will handle parsing & migrating it automatically for us.

If for some reason this function needed a legacy filter shape, either multiple types could be registered as a means of tracking version (`types: ['filter_v1']`), or the author could just migrate it backwards in the function body itself (maybe the "down" migration function would also be provided by the owner of Filters). Still not sure what the best solution is for this case.